### PR TITLE
Staff can sign in

### PR DESCRIPTION
**Pull request by erikbjoern**
[PT link](https://www.pivotaltracker.com/story/show/172936105)

Unofficial user story:
```
As an owner of the news site
In order to restrict access to creating content
I would like to require staff to sign in to the staff site in order to use it
```

## Feature
* Staff can sign in

Staff will receive their account through the admin who for now creates it in the database console.

### User model, through Devise
* Email
* Encrypted password

### Endpoint for signing in
* `post` to `/auth/sign_in`, see README for response

## Gems added:
* Devise Token Auth

### CODE REVIEW

**Review by kermit-klein — APPROVED** [review]

File: spec/factories/users.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+FactoryBot.define do
+  factory :user do
+    name { "User Example" }
+    email { "MyString@mail.com" }
```
Comment (designerofthing) [inline spec/factories/users.rb:3]: Should this not have an image param as well?

File: spec/requests/api/sessions_spec.rb [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  let(:expected_response) do
+    {
+      'data' => {
+        'id' => user.id, 'uid' => user.email, 'email' => user.email,
+        'provider' => 'email', 'allow_password_change' => false, 'name' => user.name,
+        'image' => user.image
+      }
+    }
+  end
+  describe 'with valid credentials' do
```
Comment (designerofthing) [inline spec/requests/api/sessions_spec.rb:15]: add space between end and describe

**Review by designerofthing — COMMENTED** [review]
Just a question and a space from me

**Review by jysmys — APPROVED** [review]

File: config/initializers/devise_token_auth.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# frozen_string_literal: true
+
+DeviseTokenAuth.setup do |config|
```
Comment (oliverochman) [inline config/initializers/devise_token_auth.rb:3]: this is not the correct configuration for devise token auth, take a look at the cooper documentation

File: db/migrate/20200522085432_devise_token_auth_create_users.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      t.boolean  :allow_password_change, default: false
+
+      t.datetime :remember_created_at
+
+      t.string   :confirmation_token
+      t.datetime :confirmed_at
+      t.datetime :confirmation_sent_at
+      t.string   :unconfirmed_email
+
+      t.string :name
```
Comment (oliverochman) [inline db/migrate/20200522085432_devise_token_auth_create_users.rb]: please remove name and email

**Review by oliverochman — COMMENTED** [review]
Why is a user signing in?

**Comment by erikbjoern** [comment]
> Why is a user signing in?
```
As an owner of the news site
In order to restrict access to creating content
I would like to require staff to sign in to the staff site in order to use it
```

**Review by ViaMarcus — APPROVED** [review]

**Pull request merged**